### DISCUSSION TITLE
Expiry date for TLS Certificate (basic profile)?

### DISCUSSION THREAD

**Discussion by Amphagory** [rabbitmq/tls-gen#55] [General]
When creating a TLS Certificate (basic profile), does it expire?  
If it does, if here a way to make it not expire?

**Comment by lukebakken** [comment]
> When creating a TLS Certificate (basic profile), does it expire?

https://github.com/rabbitmq/tls-gen/blob/main/basic/openssl.cnf#L22

> If it does, if here a way to make it not expire?

No, but you can modify the value of `default_days` to something far in the future.

**Reply by Amphagory** [reply]
> > When creating a TLS Certificate (basic profile), does it expire?
> 
> https://github.com/rabbitmq/tls-gen/blob/main/basic/openssl.cnf#L22
> 
> > If it does, if here a way to make it not expire?
I created certs mine like a few years ago, seems like I can use `make regen` to generate new cert (and I think I need to copy this same cert to the other RabbiMQ nodes).  Is that the process to update the certs for all RabbitMQ nodes in the cluster?

> 
> No, but you can modify the value of `default_days` to something far in the future.
I see, either I use `make DAYS_OF_VALIDITY=`  or update the openssl.cnf variable default_days to large number say 36500 (100 years).
